shim/src/evaluators.js

```diff
@@ -0,0 +1,35 @@
+import { evaluate } from './evaluate';
+import { setPrototypeOf } from './commons';
+
+function getEvalEvaluator(sandbox) {
+  const o = {
+    // trick to set the name of the function to "eval"
+    eval: function(sourceText) {
```

Comment:
We should use concise method syntax for this
```js
    eval(sourceText) {
```
so that the eval function has no [[Construct]] behavior.

Reply:
@erights is there a different technique?

```js
const o1 = { eval: function(sourceText){} };
const o2 = { eval(sourceText){} };

console.log(o1.eval.constructor); // [Function: Function]
console.log(o2.eval.constructor); // [Function: Function]
console.log(o1.eval.constructor("return 123")()); // 123
console.log(o2.eval.constructor("return 123")()); // 123
```

Reply:
By [[Construct]] behavior, I mean whether you can `new` it

```js
new eval('x');  // throws TypeError because the builtin has no [[Construct]] behavior
const o1 = { eval: function(s){} };
const o2 = { eval(s){} };
new o1.eval('x');  // a bogus instance we don't want
new o2.eval('x');  // the TypeError we want
```

Reply:
Great.